### PR TITLE
As/server not found error page

### PR DESCRIPTION
**Pull request by soncuteanca**
# Description

This test verifies that when a user navigates to a non-existent site, a 'Server Not Found' error title is displayed in the browser tab. The error page contains the correct elements, and the suggested link redirects to the appropriate page.

## Bugzilla bug ID

**ID: 1902783**
**Link: https://bugzilla.mozilla.org/show_bug.cgi?id=1902783**

## Type of change

Please delete options that are not relevant.

- [x] New Test
- [x] New POM

# How does this resolve / make progress on that bug?
Completed

Please describe the progress or significance with respect to the bug listed above.

# Screenshots / Explanations

Please upload any relevant media or add a relevant description with respect to the bug listed above.

# Comments / Concerns

Please add a short blurb about any comments or concerns that this change might cause.

### CODE REVIEW

File: tests/address_bar_and_search/test_server_not_found_error.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    # Create objects
+    nav = Navigation(driver).open()
+    tabs = TabBar(driver)
+    error_page = ErrorPage(driver)
+
+    nav.search(url)
+
+    # Verify the tab title
+
+    WebDriverWait(driver, 30).until(
```
[the remark this answers is not in the document]
Comment (ben-c-at-moz) [reply tests/address_bar_and_search/test_server_not_found_error.py:41]: Also, why do we need to wait for 30 second here?

**Review by ben-c-at-moz — APPROVED** [review]
A couple of questions, but this is great.

File: tests/address_bar_and_search/test_server_not_found_error.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+@pytest.fixture()
+def add_prefs():
+    return [
+        ("browser.search.region", "US"),
+    ]
+
+
+@pytest.mark.parametrize(
```
[the remark this answers is not in the document]
Comment (soncuteanca) [reply tests/address_bar_and_search/test_server_not_found_error.py]: Yeah, is maybe redundant, cause one site is enough to validate this test, I did a second commit using hardcoded data.

File: tests/address_bar_and_search/test_server_not_found_error.py (hunk 1 above)
Comment (soncuteanca) [reply tests/address_bar_and_search/test_server_not_found_error.py:32]: The tab loads very slowly. A shorter timeout (like 20 seconds) is not sufficient for the tab to load completely on my end, resulting in a TimeoutException.

**Pull request merged**